lumicks/pylake/kymotracker/kymoline.py

```diff
@@ -416,3 +416,49 @@ def fit_binding_times(self, n_components):
         return _kinetic_mle_optimize(
             n_components, dwelltimes_sec, min_observation_time, max_observation_time
         )
+
+    def _histogram_binding_events(self, kind, bins=10):
+        """Make histogram of bound events.
+
+        Parameters
+        ----------
+        kind : {'binding', 'all'}
+            Type of events to count. 'binding' counts only the first position of each track whereas
+            'all' counts positions of all time points in each track.
+        bins : int or sequence of scalars
+            Definition of the histogram bins; passed to `np.histogram()`. If `int`, defines the total
+            number of bins. If sequence of scalars, defines the edges of the bins.
+        """
```

Comment:
This sounds like there are two functions packed into one. We could maybe split this into `_histogram_count_first_positions()` and a `_histogram_count_all_points()` if that makes sense.

Reply:
I had a chat about this with Tiago to try to get some insight about why this might be a better option; for something like C++ I can see that there might be compiler optimizations or other reasons to split this into separate functions. But for python I don't think it makes sense. We'd end up with two functions that are identical except for a single line, so I'd rather take the de-duplication and deal with a bit more logic within the single function.

That being said, I'm not particularly fond of using a string as an argument to determine the type of histogram calculated; but this is a fairly common pattern in python so I guess it's ok - I don't think there's really a better option, especially considering that this might become a public function in the future. I would've also preferred something like `type` instead of `kind`, but `type` is a core function. I'm open to other ideas for the argument name though